client.go

```diff
@@ -198,7 +217,16 @@ type newRequestError struct {
 func NewClient(config Config) (Marathon, error) {
 	// step: if no http client, set to default
 	if config.HTTPClient == nil {
-		config.HTTPClient = http.DefaultClient
+		config.HTTPClient = defaultHttpClient
+	}
+
+	if config.HTTPSSEClient == nil {
+		config.HTTPSSEClient = defaultHttpSSEClient
+	} else if config.HTTPSSEClient.Timeout != 0 {
+		return nil, fmt.Errorf(
+			"HTTPSSEClient.Timeout cannot be set (got %v)",
```

Comment:
I'd rephrase this slightly less technical as 

"Global timeout must not be set on custom HTTP client for SSE connections (got %s)"